src/include/cell.i

```diff
@@ -988,8 +987,17 @@ __cell_unpack_window_cleanup(WT_SESSION_IMPL *session, const WT_PAGE_HEADER *dsk
         if (ta->newest_stop_txn != WT_TXN_MAX) {
             ta->newest_stop_txn = WT_TXN_NONE;
             F_SET(unpack_addr, WT_CELL_UNPACK_TIME_WINDOW_CLEARED);
-            if (ta->newest_stop_ts == WT_TS_MAX)
+
+            /*
+             * The newest_stop_ts is WT_TS_MAX when its newest_stop_txn is not WT_TXN_MAX is
+             * possible only for the non-timestamped tables. WT_TS_MAX is not considered for
```

Comment:
The first sentence of both of these comment was a little hard for me to understand.  Might I suggest:
The combination of newest stop timestamp being WT_TS_MAX while the newest stop transaction not being WT_TXN_MAX is possible only for the non-timestamped tables.

Reply:
Thanks. Fixed as you suggested.